app/samples/phase_detection/src/main.cpp

```diff
@@ -0,0 +1,241 @@
+    }
+    controller.SubmitEvent(Sources::IMU1, Events::PadReady);
+    controller.WaitUntilEvent(Events::PadReady);
+
+    while (true) {
+        k_timer_status_sync(&imu_timer);
+        uint32_t timestamp = k_uptime_get();
+        bool good = acc.UpdateSensorValue();
+        if (!good) {
+            LOG_ERR("Failure reading imu");
+        }
+        double x = acc.GetSensorValueDouble(SENSOR_CHAN_ACCEL_X);
+        double y = acc.GetSensorValueDouble(SENSOR_CHAN_ACCEL_Y);
+        double z = acc.GetSensorValueDouble(SENSOR_CHAN_ACCEL_Z);
+        double mag = sqrt(x * x + y * y + z * z);
+
+        // Boost Detecting
+        if (!controller.HasEventOccured(Events::Boost)) {
+            boost_detector.feed(timestamp, mag);
+            coast_detector.feed(timestamp, mag);
+            if (boost_detector.passed()) {
+                controller.SubmitEvent(Sources::IMU1, Events::Boost);
+            }
+        } else if (!controller.HasEventOccured(Events::Coast)) {
+            coast_detector.feed(timestamp, mag);
+            if (coast_detector.passed()) {
+                controller.SubmitEvent(Sources::IMU1, Events::Coast);
+            }
+        }
+    }
+}
+K_THREAD_DEFINE(imu_thread, 1024, imu_thread_f, NULL, NULL, NULL, 0, 0, 0);
+
+using SampleType = LinearFitSample<double>;
+static constexpr std::size_t window_size = 10;
+using SummerType = RollingSum<SampleType, window_size>;
+
+struct Line {
+    double m;
+    double b;
+    constexpr bool operator==(const Line &rhs) const { return m == rhs.m && b == rhs.b; }
+};
+// Value returned when we don't have enough information to fit a line
+constexpr Line bad_line{0, 0};
+
+Line find_line(const SummerType &summer) {
+    std::size_t N = summer.size();
+    SampleType E = summer.sum();
+    double denom = (N * E.xx - E.x * E.x);
+    if (denom == 0) {
+        // printf("Would have divided by 0\n");
+        return bad_line;
+    }
+    double m = (N * E.xy - E.x * E.y) / denom;
+    double b = (E.y - m * E.x) / N;
+    return Line{m, b};
+}
+
+double l_altitude_conversion(double pressure_kpa, double temperature_c) {
+    double pressure = pressure_kpa * 10;
+    double altitude = (1 - pow(pressure / 1013.25, 0.190284)) * 145366.45 * 0.3048;
+    return altitude;
+}
+
+using BoostDebouncerT = Debuouncer<ThresholdDirection::Over, double>;
+using NoseoverDebouncerT = Debuouncer<ThresholdDirection::Under, double>;
+using MainHeightDebouncerT = Debuouncer<ThresholdDirection::Under, double>;
+using NoVelocityDebouncerT = Debuouncer<ThresholdDirection::Under, double>;
+
+void barom_thread_f(void *, void *, void *) {
+    CBarometer barometer(*DEVICE_DT_GET_ONE(openrocket_barometer));
+    if (!barometer.IsReady()) {
+        LOG_WRN("Accelerometer not ready");
+    }
+    controller.SubmitEvent(Sources::Barom1, Events::PadReady);
+    controller.WaitUntilEvent(Events::PadReady);
+
+    SummerType velocity_summer{SampleType{0, 0}};
+
+    constexpr uint32_t boost_time = 2000;   // ms
```

Comment:
constexpr for fun but in reality we could load these from settings api or something home made so that we can initialize with configurable data